### PR TITLE
Add install command support for package version

### PR DESCRIPTION
**Pull request by realies**
Perhaps also replace `packageName` to `json.name` (if available) after line 33.

### CODE REVIEW

**Comment by astorije** [comment]
Uh oh! @realies, do you need any help with this?

**Comment by astorije** [comment]
Alright so after chatting on IRC, documenting here steps to clean this PR in case it helps.

You'll notice that your PR says: "This branch has conflicts that must be resolved - Conflicting files: `src/command-line/install.js`", and the changed files show changes that aren't yours. In fact, those changes are the reason of the conflicting files.

To fix this, you need to change your commits so it only contains your stuff. Note that all that follows has to happen in your terminal, not on the GitHub UI, so make sure you are in a terminal, you ran `git pull` on your own repo and your own branch. Then, to fix this:

- The `master` branch in your fork be outdated from ours. Let's start by syncing those:
  - On your local repo, start by telling your git what is the point of reference, i.e. our upstream repo: `git remote add upstream https://github.com/thelounge/thelounge`
  - Make sure you don't have local changes (`git status` must be empty) and that you are in your `master` branch: `git checkout master`
  - Sync it with ours: `git fetch upstream && git rebase upstream/master` 
  - `git status` should tell you your local `master` is ahead of a few commits, run `git push` (don't `--force` it as it might be a sign that something is wrong) to update your fork.
- Now, let's clean up your branch:
  - Go to your branch: `git checkout patch-4`
  - Sync it with `master`: `git rebase master`
  - 💥 Conflicts! (Probably)
  - In your situation, I would run `git reset HEAD src/command-line/install.js`, and clean up things manually.
  - When you're done cleaning things, tell `git` you're done fixing conflicts: `git add src/command-line/install.js && git rebase --continue`
  - Check that your commit is now looking good with `git show`
  - If everything looks alright, update your branch on GitHub, and therefore this PR: `git push --force` (this time you are going to need to `--force` it)

Let me know how this goes!

**Comment by realies** [comment]
@astorije, the `git magic` guide is much appreciated. Saving for future reference.

File: src/command-line/install.js [hunk 1]
```diff
@@ -71,7 +76,7 @@ program
 					"--non-interactive",
 					"--cwd",
 					packagesPath,
-					`${packageName}@${json.version}`,
+					`${json.name}@${json.version}`,
```
[the remark this answers is not in the document]
Comment (McInkay) [reply src/command-line/install.js]: We used to support it. Then it was removed. Pretty sure I mentioned this at the time. Will check when letter.

[the remark this answers is not in the document]
Comment (astorije) [reply src/command-line/install.js]: Yeah, so looking at [this thread](https://github.com/thelounge/thelounge/pull/2063), it looks like I was confused by this change, but I _think_ I understand it now 😅

[the remark this answers is not in the document]
Comment (xPaw) [reply src/command-line/install.js]: It's used to specify the exact version that exists in the NPM registry. Before this PR it looked up `latest` via the API, and put that version in the command. This PR adds support for installing other versions or tags, as they are all looked up via the api.

File: src/command-line/install.js [hunk 2]
```diff
@@ -23,16 +23,21 @@ program
 
 		log.info("Retrieving information about the package...");
 
+		const split = packageName.split("@");
+		packageName = split[0];
+		const packageVersion = split[1] || "latest";
+
 		packageJson(packageName, {
 			fullMetadata: true,
+			version: packageVersion,
```
[the remark this answers is not in the document]
Comment (realies) [reply src/command-line/install.js]: @astorije, https://github.com/sindresorhus/package-json/issues/42#issuecomment-371387548

[the remark this answers is not in the document]
Comment (astorije) [reply src/command-line/install.js]: Thanks @realies!

**Pull request merged**